docs/oauth-dev.md

```diff
@@ -0,0 +1,121 @@
+---
+title: OAuth for Plugins
+---
+**IMPORTANT:** OAuth for plugins is available in Tableau 2021.1 and newer. Multi-IDP config for plugins is available starting in Tableau 2023.1.
+
+This document explains how to add OAuth to a connector. It's meant for developers. For an explanation of OAuth configuration and usage see [OAuth Config](./oauth.md).
+
+**In this section**
+
+* TOC
+{:toc}
+
+# How to Enable OAuth for a Plugin Connector
+
+First check your database and driver documentation to make sure it supports OAuth. 
+
+To enable OAuth for your connector add an `<oauth-config>` field in the manifest.xml and link to an oauthConfig.xml you created, described below.
+
+For a complete example please refer to https://github.com/tableau/connector-plugin-sdk/tree/master/samples/scenarios/snowflake_oauth.
+
+```xml
+  manifest.xml
+
+  <?xml version='1.0' encoding='utf-8' ?>
+  <connector-plugin class=...>
+    ...
+    <dialect file='dialect.tdd'/>
+    <oauth-config file='oauthConfigPing.xml'/>    <!-- add this to to define your OAuth Configs -->
+  </connector-plugin>
+```
+
+Starting in Tableau 2023.1, you can add multiple OAuth configs, embedded in the plugin. The end users may also provide external/custom OAuth configurations:
```

Comment:
here we can consider linking to ## Multiple Embedded OAuth Configs portion? 

Also this part is sorta duplicated with this part from oauth page? 
```This file may be provided in the following ways:
- Bundled in the connector package (only available to plugin authors)
- Installing on a client machine for Tableau Desktop, Prep, or Bridge
- Installing for a site on Tableau Cloud or Tableau Server
```

and "The end users may also provide custom OAuth configurations: " i think we should remove word "external".